elixir/apps/domain/lib/domain/accounts/account/changeset.ex

```diff
@@ -2,22 +2,61 @@ defmodule Domain.Accounts.Account.Changeset do
   use Domain, :changeset
   alias Domain.Accounts.Account
 
-  def create_changeset(attrs) do
-    %Account{}
+  def changeset(account, attrs) do
+    account
     |> cast(attrs, [:name, :slug])
     |> validate_required([:name])
-    |> validate_length(:name, min: 1, max: 255)
+    |> validate_name()
+    |> trim_name_whitespace()
+    |> set_slug_default()
+    |> downcase_slug()
+    |> remove_spaces_from_slug()
     |> validate_slug()
-    |> validate_length(:slug, min: 3, max: 100)
+    |> unique_constraint(:slug, name: :accounts_slug_index)
+  end
+
+  def create_changeset(attrs) do
+    %Account{}
+    |> changeset(attrs)
+  end
+
+  defp validate_name(changeset) do
+    changeset
+    |> validate_length(:name, min: 3, max: 255)
+    |> validate_format(:name, ~r/^[a-zA-Z0-9\s_-]+$/,
```

Comment:
I think the name can be pretty much any string, doesn't matter what it contains (it's what you will see in UI, eg. "Firezone, Inc."). 255 is too long though for it, maybe let's set it to 64 and increase if requested?

Reply:
This makes sense now too.  This is similar to the comment above when we were planning on making the `slug` out of the `name` I wanted to make sure it was easy to convert one to the other, but if we're going to have them completely separate we should let people display whatever name they want.

Reply:
This has been updated to only validate length from `min:3` to `max:64`